ethereum_history_api/circuits/lib/src/misc/fragment.nr

```diff
@@ -25,4 +31,50 @@ impl<MAX_DATA_LEN> Fragment<MAX_DATA_LEN> {
 
         Fragment { offset: 0, length: self.length, data: new_data }
     }
+
+    pub fn is_empty(self) -> bool {
```

Comment:
`is_empty()`

Reply:
We need self as an argument here to use self.length